katsdpimager/frontend.py

```diff
@@ -459,6 +495,8 @@ def process_channel(dataset, args, start_channel,
             restore_image.copy_region(queue, model, (), np.s_[pol])
         queue.finish()
 
+    if grid_p.beams:
+        model /= pbeam
```

Comment:
Because you said I should do the primary beam correction as the very last step :-) I think convolving corrected_model with the restoring beam might be more correct, because total flux in the final image should then match the flux in the model (which won't quite be true currently where the beam changes sharply).